core/src/main/java/com/linecorp/armeria/common/CoreBlockhoundIntegration.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common;
+
+import java.util.ResourceBundle;
+import java.util.UUID;
+
+import com.linecorp.armeria.common.annotation.UnstableApi;
+
+import reactor.blockhound.BlockHound.Builder;
+import reactor.blockhound.integration.BlockHoundIntegration;
+
+/**
+ * A {@link BlockHoundIntegration} for the core module.
+ */
+@UnstableApi
+public final class CoreBlockhoundIntegration implements BlockHoundIntegration {
+    @Override
+    public void applyTo(Builder builder) {
+        // short locks
+        builder.allowBlockingCallsInside("com.linecorp.armeria.client.HttpClientFactory",
+                                         "pool");
+        builder.allowBlockingCallsInside("com.linecorp.armeria.internal.common.util.ReentrantShortLock",
+                                         "lock");
+
+        // Thread.yield can be eventually called when PooledObjects.copyAndClose is called
+        builder.allowBlockingCallsInside("io.netty.util.internal.ReferenceCountUpdater", "release");
+
+        // hdr histogram holds locks
+        builder.allowBlockingCallsInside("org.HdrHistogram.ConcurrentHistogram", "getCountAtIndex");
+        builder.allowBlockingCallsInside("org.HdrHistogram.WriterReaderPhaser", "flipPhase");
+
+        // StreamMessageInputStream internally uses a blocking queue
+        // ThreadPoolExecutor.execute internally uses a blocking queue
+        builder.allowBlockingCallsInside("java.util.concurrent.LinkedBlockingQueue", "offer");
```

Comment:
Q: If Blockhound is installed with this integration, does this rule affect your codebase? If this rule is only applied to Armeria codebase, it is fine.

Reply:
I believe it's fine because it holds a lock for a short period of time:
```java
public boolean offerLast(E e) {
    if (e == null) throw new NullPointerException();
    Node<E> node = new Node<E>(e);
    final ReentrantLock lock = this.lock;
    lock.lock();
    try {
        return linkLast(node);
    } finally {
        lock.unlock();
    }
}
```

Reply:
> Q: If Blockhound is installed with this integration, does this rule affect user’s codebase? If this rule is only applied to Armeria codebase, it is fine.

It will be applied to the entire user's codebase.

If this is really a problem I think it is possible to divide `CoreBlockhoundIntegration` into two parts so that users can choose to opt-out of the non-armeria related rules if they want.

Reply:
I agree that it makes sense to ignore `LinkedBlockingQueue.offer` method.
Let's keep it as is.